File: JS/01.Basics/99.Pre-Exam/03.FinalCompetition.js
function calculateMoney(input) {

    let dancersCount = parseInt(input[0]);
    let points = parseFloat(input[1]);
    let season = input[2];
    let location = input[3];
    let winnings = dancersCount * points;

    if (location === "Abroad") {
        winnings += 0.5 * winnings;
    }

    let expensesPercentage = 0;
    if (season === "summer" && location === "Bulgaria") {
        expensesPercentage = 0.05;
    } else if (season === "winter" && location === "Bulgaria") {
        expensesPercentage = 0.08;
    } else if (season === "summer" && location === "Abroad") {
        expensesPercentage = 0.10;
    } else if (season === "winter" && location === "Abroad") {
        expensesPercentage = 0.15;
    }

    let expenses = expensesPercentage * winnings;
    let remainingMoney = winnings - expenses;

    let charity = 0.75 * remainingMoney;
    let moneyPerDancer = (remainingMoney - charity) / dancersCount;

    charity = charity.toFixed(2);
    moneyPerDancer = moneyPerDancer.toFixed(2);

    console.log(`Charity - ${charity}`);
    console.log(`Money per dancer - ${moneyPerDancer}`);
}

calculateMoney(["1", "89.5", "summer", "Abroad"]);
calculateMoney(["25", "98", "winter", "Bulgaria"]);

File: Python/01.ProgrammingBasics/99.Pre-Exam/03.FinalCompetition/_03.FinalCompetition.py
dancers = int(input())
points = float(input())
season = input()
location = input()

prize_money = dancers * points

if location == "Abroad":
    prize_money += 0.50 * prize_money

if season == "summer":
    expenses_percentage = 0.10 if location == "Abroad" else 0.05
else:
    expenses_percentage = 0.15 if location == "Abroad" else 0.08

prize_money -= expenses_percentage * prize_money

charity_money = 0.75 * prize_money
remaining_money = prize_money - charity_money
money_per_dancer = remaining_money / dancers

print(f"Charity - {charity_money:.2f}")
print(f"Money per dancer - {money_per_dancer:.2f}")
